mantidproject/mantid
# Engineering Diffraction GUI move Algorithm and ADS code to model

As part of #21171, all code related to Mantid algorithms and the ADS has been moved from `EnggDiffFittingPresenter` to the new `EnggDiffFittingModel`, giving a more decoupled system.

This PR was originally also meant to take care of enabling fitting of multiple runs at the same time, but the refactoring job was sufficiently large that I thought it best to make an intermediate PR here (apologies for the huge diff).

Please note: the unit test at `EnggDiffFittingPresenterTest.h:428` is disabled, as the functionality is not yet properly implemented. This will be re-enabled and fixed when **Fit all** is finally complete.

**To test:**

Some sample data is [here](https://github.com/mantidproject/mantid/files/1460806/ENGINX_focused_runs.zip).

Load a run from the above data by browsing with the first **Browse** button, then clicking **Load**.
Select the loaded run from the list widget by clicking its label - it should be plotted in the plot window.
Click **Fit** to do a fit - some workspaces with fit parameters should appear in the Workspaces area, and the fitted peaks should be plotted over the original plot.

Try changing the peaks with the Peaks line-edit, fitting while multiple runs are loaded, and anything else you think might break things.

Check that fitting information was saved (as `.csv`). On Windows this will be in `C:\EnginX_Mantid\User`, on Linux `~/EnginX_Mantid/User`.

Partially addresses #21171 #21082 

**Release Notes** 
The change is not complete - I will add it to the release notes once multi-run fitting has been implemented.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
